### ISSUE TITLE
A11y_.NET Core_WPF_ExpenseItIntro_ExpenseReportfor_Keyboard: The table column headers are not accessible by keyboard.

### ISSUE THREAD

**Issue by Swati1700** [dotnet/wpf#6737]
“[Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

**GitHubTags:**#A11yMAS;#A11yTCS;#.NET Core[WPF]-Win32-June2022;#.NET Core;#A11ySev2;#WCAG2.1.1;#DesktopApp;#Win11;#Keyboard;#Benchmark;

### Environment Details:
.NET Core
WPF
Operating System: Windows 11 Enterprise 21H2

### Steps to Reproduce:

1. Launch VS 2019 Int Preview
2. Navigate to Getting started and right click on the Expense it intro and click on build
3. Then right click on Expense it Intro then Click on Debug-->Start New Instance. 
4. Navigate to Names list and select any name and click on View.
5. 'Expense Report for' screen will open
6. Tab till table and observe the issue

### Actual:
The table column headers "ExpenseType" and "Amount" are not accessible by keyboard. While using mouse and clicking on this column headers sorting functionality is performed but we are not able to use sorting functionality by using keyboard.

### Expected:
The table column headers "ExpenseType" and "Amount" should be accessible by keyboard. So that the sorting functionality should be accessible by using keyboard.

### User Impact:
Keyboard users will not be able to sort the data by using keyboard.
[WPF keyboard.zip](https://github.com/dotnet/wpf/files/9008397/WPF.keyboard.zip)

**Comment by anjalisheel-wpf** [comment]
@Swati1700 This is working as per the design.

**Comment by Swati1700** [comment]
@anjalisheel-wpf Here the issue is while navigating to the column headers by using mouse we are able sort the table data after click on column header but here we are unable to navigate column headers so by using keyboard, so we are not able to sort the table data. That's why the column headers should be keyboard accessible. 
If it is by designed then please provide proper reason.

**Comment by merriemcgaw** [comment]
@anjalisheel-wpf is there a keyboard command to sort by the active column? I know that WinForms for the DataGridView has that option. Does WPF expose some keyboard way of initiating sort on a column?

**Comment by pchaurasia14** [comment]
The fix has already been merged into the codebase. 
This issue can be closed post testing on the latest installer.

cc- @Swati1700  @merriemcgaw

**Comment by merriemcgaw** [comment]
@Swati1700 this is ready for re-testing. Note that rather than making the headers available WPF is starting by creating a keyboard shortcut to sort by the current column. This will be documented and allows the keyboard user to accomplish the same thing that the mouse user does but by using a keyboard command instead. @dipeshmsft - what was the keyboard shortcut again?

**Comment by dipeshmsft** [comment]
The shortcut for doing this is F3

**Comment by merriemcgaw** [comment]
@rperez030 - I want to confirm that this is an acceptable solution to solve this issue. WinForms was given that leeway a couple years ago and we implemented the same thing - allowing a user to sort the current column with the F3 key. If this is not an acceptable solution, let us know ASAP, there isn't much time to get a full feature implementation done in .NET 7 so I'd need to start the exception process yesterday 😺

**Comment by rperez030** [comment]
@merriemcgaw  I agree this is an acceptable solution as long as it is documented, and developers know to make that shortcut discoverable in their applications. It meets MAS 2.1.1 by making the functionality available through keyboard and matches the existing behavior in winforms. @Swati1700  if you can verify that the fix works as described, I think the issue can be closed.

**Comment by miloush** [comment]
F3 is often assigned to search/find next: 

- Search for a file or folder in File Explorer. [Windows](https://support.microsoft.com/en-gb/windows/keyboard-shortcuts-in-windows-dcc61a57-8ff0-cffe-9796-cb9706c75eec)
- Search for the next instance of the text in the Finddialog box [Apps/WordPad](https://support.microsoft.com/en-gb/windows/keyboard-shortcuts-in-apps-139014e7-177b-d1f3-eb2e-7298b2599a34)
- Search for an item. [Outlook](https://support.microsoft.com/en-us/topic/3cdeb221-7ae5-4c1d-8c1d-9e63216c1efd?ui=en-us&rs=en-gb&ad=gb)
- While searching the current page, move to the next result. [OneNote](https://support.microsoft.com/en-us/topic/44b8b3f4-c274-4bcc-a089-e80fdcc87950?ui=en-us&rs=en-gb&ad=gb)
- Display the Find and Replace task pane, with the Find option selected [Publisher](https://support.microsoft.com/en-us/office/keyboard-shortcuts-in-publisher-289f9c39-5584-49f0-b9a9-081e646152ea?ui=en-us&rs=en-gb&ad=gb)
- Find next [Visual Studio](https://docs.microsoft.com/en-gb/visualstudio/ide/default-keyboard-shortcuts-in-visual-studio?view=vs-2022)
- Find Next in Script [PowerShell](https://docs.microsoft.com/en-us/powershell/scripting/windows-powershell/ise/keyboard-shortcuts-for-the-windows-powershell-ise?view=powershell-7.2)

What is the expected behavior when user is in a table and the application already has F3 key assigned?

How does it affect sorting when the table is sorted by multiple columns?

**Comment by merriemcgaw** [comment]
The multi-column sort is something that needs to be figured out over the longer .NET 8 timeframe. I do agree that F3 is often associated with Find. Perhaps in .NET 8 we can extend this feature to make the sort key configurable or allow direct access to the column headers. This is an acceptable solution for now simply because it's consistent with WinForms, so .NET Desktop based apps are aligned. 

As for how the app responds to F3 when it's already assigned - we'd have to talk to @dipeshmsft for how he implemented it. I'd recommend that we let the app override the default setting and allow the developer to specify an alternate keyboard command for their application. That way F3 can be the default and developers have flexibility to set what they need in their app. Given the short runway in .NET 7 I propose to leave this as is, and work with the WPF team to flesh out a full spec of what would be a better long term approach in .NET 8, building on the work done in .NET 7.

**Comment by miloush** [comment]
The PR is for key down event on DataGridCell, so it will get handled before any parent/window gets a chance (unless they use the preview events), i.e. a potentially breaking change for existing applications and new applications will have to avoid this decades old shortcut for searching, confusing users.

The answer to multi-column case here is the same as when user clicks on the column header, which sounds reasonable to me, though from accessibility point of view you want to have more control over it.

> I propose to leave this as is, and work with the WPF team to flesh out a full spec 

Sure, I realize I am late here, I just wish with all these issues that there was a full spec first before we start introducing random breaking changes.

**Comment by merriemcgaw** [comment]
@Swati1700 please validate this bug and close ASAP. @rperez030 has approved the keyboard command to do sorting based on the current column, so we don't have to specifically focus on the headers if we can accomplish the same thing with the keyboard by focusing on a cell who's column we want to sort by.

**Comment by pchaurasia14** [comment]
@Swati1700 - Please retest this issue with .NET 7 RC1.

**Comment by Swati1700** [comment]
@pchaurasia14 We have retested but we are getting error. Could you please provide the .exe file to verify the bug. Thanks

**Comment by singhashish-wpf** [comment]
@Swati1700 The fix is in product. Have you installed .NET7 latest installer and are you checking with that version of dotnet?

**Comment by Swati1700** [comment]
Verified the issue and bug is fixed. Hence closing the bug.

https://user-images.githubusercontent.com/86710888/192529609-6a9f8c4d-15a4-4f38-b1f3-ad4895cd0bed.mp4
